src/display.jl

```diff
@@ -437,55 +444,26 @@ end
 
 If you want a more tweakable interface, consider using [`VideoStream`](@ref) and
 [`save`](@ref).
-
-## Examples
-
-```julia
-scene = lines(rand(10))
-record(scene, "test.gif") do io
-    for i in 1:255
-        scene.plots[:color] = Colors.RGB(i/255, (255 - i)/255, 0) # animate scene
-        recordframe!(io)
-    end
-end
-```
-or
-```julia
-scene = lines(rand(10))
-record(scene, "test.gif", 1:255) do i
-    scene.plots[:color] = Colors.RGB(i/255, (255 - i)/255, 0) # animate scene
-end
-```
 """
-function record(func, scene, path; framerate::Int = 24)
+function record(func, scene, path; framerate::Int = 24, kwargs...)
     io = VideoStream(scene; framerate = framerate)
     func(io)
-    save(path, io; framerate = framerate)
+    save(path, io; framerate = framerate, kwargs...)
 end
 
-"""
-    record(func, scene, path, iter; framerate = 24)
-
-This is simply a shorthand to wrap a for loop in `record`.
-
-Example:
+const _SLEEP_WHEN_RECORDING = Ref(true)
 
-```example
-    scene = lines(rand(10))
-    record(scene, "test.gif", 1:100) do i
-        scene.plots[:color] = Colors.RGB(i/255, 0, 0) # animate scene
-    end
-```
-"""
-function record(func, scene, path, iter; framerate::Int = 24)
+function record(func, scene, path, iter; framerate::Int = 24, sleep = _SLEEP_WHEN_RECORDING[], kwargs...)
```

Comment:
Whats this `_SLEEP_WHEN_RECORDING` thing?

Reply:
I couldn't think of a better way to allow the default value of `sleep` to be changed, so I made it a Ref.  Thinking about it now, we should probably use an env variable instead...

Reply:
What's the use case to change it globally?

Reply:
MakieGallery mainly - we want users to see the animation if they run it, but we also want to record it quickly on CI.

Reply:
If we want a mode, that records as fast as possible without adhering to a framerate, we should add that to the API and use that in `record` in our examples!

Reply:
Should we make this the default?

Reply:
Sorry to be so naggy, but messing up our API with confusing hacks like this is simply not worth the short term gain ;) How is any outside collaborator able to interpret this ENV variable when trying to update the code? How would anyone figure out how to replace this ENV in a refactor that tries to make the API more consistent? 
We really need to design things with maintaining & documenting our API as the main bottleneck in mind, instead of sacrificing the former for improving our CI, or getting a quick hack in to allow high dpi support^^ 
I'm talking from experience here, since I'm the biggest offender of this principle, and everytime I did something like this, I got really sad 1 year later when trying to refactor the CODE and clean up all the hacks.

Reply:
Fair enough :) - this solution is pretty hacky no matter which way you slice it.

Reply:
Also, it doesn't look like this impacts CI time too much, so we could probably skip updating the Gallery...